NExtends/Expressions/ExpressionHelper.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace NExtends.Expressions
+{
+	public class ExpressionHelper
+	{
+		// Utile pour les types créés à partir de Group By d'un autre type
+		public static Expression<Func<TOut, bool>> Transfert<TIn, TOut>(Expression<Func<TIn, bool>> expression)
+		{
+			return TransfertWithRename<TIn, TOut>(expression, new Dictionary<string, string>());
+		}
+		/// <summary>
+		/// Permet de transférer un délégué d'un enfant vers son parent étant connu le nom de la propriété qui permet de passer du parent à l'enfant
+		/// </summary>
+		/// <typeparam name="TChild">Le type enfant</typeparam>
+		/// <typeparam name="TParent">Le type parent</typeparam>
+		/// <param name="expression">L'expression (MemberExpression en général) sur l'enfant</param>
+		/// <param name="childName">Le nom de la propriété permettant d'accéder à l'enfant depuis le parent</param>
+		/// <returns></returns>
+		public static LambdaExpression TransfertToParent<TChild, TParent>(LambdaExpression expression, string childName)
+		{
+			var param = Expression.Parameter(typeof(TParent), "p");
+			var body = Expression.MakeMemberAccess(param, typeof(TParent).GetProperty(childName));
+
+			Expression sub = Expression.PropertyOrField(body, ((PropertyInfo)(expression.Body as MemberExpression).Member).Name);
+
+			return Expression.Lambda(sub, param);
+		}
+		public static Expression<Func<TOut, bool>> TransfertWithRename<TIn, TOut>(Expression<Func<TIn, bool>> expression, Dictionary<string, string> propertyRenaming)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Linq.Expressions.LambdaExpression' instead of 'System.Linq.Expressions.Expression<System.Func<TIn, bool>>'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3242)